Sources/App/Core/Mastodon.swift

```diff
@@ -18,34 +18,45 @@ import Vapor
 
 enum Mastodon {
 
-    private static let instance = "mas.to"
-    private static let apiURL = "https://\(instance)/api/v1/statuses"
+    private static let apiHost = "mas.to"
+    private static let apiPath = "/api/v1/statuses"
     static let postMaxLength = 490  // 500, leaving some buffer for unicode accounting oddities
 
     struct Credentials {
         var accessToken: String
     }
 
-    // NB: _testEncodedURL is a callback that exists purely to be able to regression test the encoded value
-    static func post(client: Client, message: String, _testEncodedURL: (String) -> Void = { _ in }) async throws {
+    static func apiURL(with message: String) throws -> String {
```

Comment:
This change allowed us to get rid of this ugly hack with the trailing closure that was purely used for testing.